lib/galaxy/jobs/runners/drmaa.py

```diff
@@ -357,16 +359,12 @@ def store_jobtemplate(self, job_wrapper, jt):
 
     def external_runjob(self, external_runjob_script, jobtemplate_filename, username):
         """ runs an external script the will QSUB a new job.
-        The external script will be run with sudo, and will setuid() to the specified user.
+        The external script needs to be run with sudo, and will setuid() to the specified user.
         Effectively, will QSUB as a different user (then the one used by Galaxy).
         """
-        script_parts = external_runjob_script.split()
-        script = script_parts[0]
-        command = [ '/usr/bin/sudo', '-E', script]
-        for script_argument in script_parts[1:]:
-            command.append(script_argument)
-
+        command = shlex.split( external_runjob_script )
         command.extend( [ str(username), jobtemplate_filename ] )
+        command.append( "--assign_all_groups"  )
```

Comment:
You can add this within the galaxy.ini, so I would prefer to not have this hardcoded here.